spec/mini_scheduler/manager_spec.rb

```diff
@@ -208,6 +208,58 @@ def perform
       manager.stop!
     end
 
+    it 'should recover from redis readonly within same manager instance' do
+      info = manager.schedule_info(Testing::SuperLongJob)
+      info.next_run = Time.now.to_i - 1
+      info.write!
+
+      manager.tick
+
+      wait_for do
+        manager.schedule_info(Testing::SuperLongJob).prev_result == "RUNNING"
+      end
+
+      # Simualate redis failure while job is running
+      MiniScheduler::ScheduleInfo.any_instance.stubs(:write!).raises(Redis::CommandError)
+
+      runner = manager.instance_variable_get(:@runner)
+      worker_threads = runner.instance_variable_get(:@threads)
+      worker_thread_ids = runner.worker_thread_ids
+
+      # Now that Redis is broken, simulate the 'SuperLongJob' ending
+      worker_threads.each(&:wakeup)
+
+      # Wait until the worker dies due to the redis failure
+      wait_for(timeout: 5) { worker_threads.reject(&:alive?).count == 1 }
+
+      # Observe the broken status
+      expect(manager.schedule_info(Testing::SuperLongJob).prev_result).to eq("RUNNING")
```

Comment:
A bit hard to follow here "observe broken status" and status is RUNNING?

Reply:
I guess missing a "current status" concept?

Reply:
Yeah '#prev_result' returns whatever the current status is in Redis. `#result` and `#result=` are used to access/set the value which will be saved on the next `#write` call (which we're not doing in the specs).

I'll update the comment to try and make this particular case  a little clearer.